### DISCUSSION TITLE
Add `<py-script-editor>` element

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pyscript/pyscript#1823] [Proposals]
I would like to enable PyData ecosystem and Python users in general to easily add a pyscript editor or playground to their site.

Streamlit, Gradio, Shiny etc. all do this. But differently and not something that others can embed in their sites. Currently I'm working on a Panel Playground in https://github.com/holoviz/panel/pull/5769. Then I thought it would be much more general and useful if pyscript provided this.

As inspiration you can take https://www.gradio.app/playground. Which I like for its simplicity and features.

It would need to

- Support hot reloading. Not reloading pyodide, packages etc. everytime to code changes. Something as fast and efficient as my Panel POC.
- Work also inside Web components/ Panel `ReactiveHTML`. Currently pyscript does not.

To support my needs.

## Panel POC

https://github.com/pyscript/pyscript/assets/42288570/43c584cb-a7b9-4ae5-be66-a3c56008bb58

## Gradio Playground

![image](https://github.com/pyscript/pyscript/assets/42288570/46903dd1-ee74-4316-8499-fd3fde87d529)

**Comment by WebReflection** [comment]
same question: which version of PyScript are you using?

**Reply by MarcSkovMadsen** [reply]
The one Panel is pointing to. With Panel 1.3.0 it is 2023.03.1.
